### PR TITLE
[chip-tool] remove extra blank line in cert print

### PR DESCRIPTION
**Pull request by chshu**
Fixes https://github.com/CHIP-Specifications/chip-test-plans/issues/2434

It's a corner case when the cert length is a particular value, which ends at a complete line. **chip-tool** prints an extra blank line:
image

After the fix, the output looks good:
image

### CODE REVIEW

File: examples/common/tracing/decoder/logging/ToCertificateString.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ const char * ToCertificate(const chip::ByteSpan & source, chip::MutableCharSpan
         return destination.data();
     }
 
-    size_t base64DataLen = BASE64_ENCODED_LEN(source.size()) + 1;
+    size_t base64DataLen = BASE64_ENCODED_LEN(source.size());
```
Comment (bzbarsky-apple) [inline examples/common/tracing/decoder/logging/ToCertificateString.cpp:43]: Isn't the +1 there needed to account for the null-terminator?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/common/tracing/decoder/logging/ToCertificateString.cpp (hunk 1 above)
Comment (chshu) [reply examples/common/tracing/decoder/logging/ToCertificateString.cpp:43]: @bzbarsky-apple It will always add a `\n` after each line or the end of the string : [ToCertificateString.cpp#L81](https://github.com/project-chip/connectedhomeip/blob/master/examples/common/tracing/decoder/logging/ToCertificateString.cpp#L81), null-terminator could indicate the end of the string, but don't need to be counted into the total length.

Otherwise, if the `\0` is the last character left, it will print an extra blank line, and cause the parsing issue.

Comment (bzbarsky-apple) [reply examples/common/tracing/decoder/logging/ToCertificateString.cpp:43]: What I mean is that this code:
```
    str.Alloc(base64DataLen);
    auto encodedLen       = chip::Base64Encode(source.data(), source.size(), str.Get());
    str.Get()[encodedLen] = '\0';
```
will write outside the buffer if `encodedLen` ends up being `BASE64_ENCODED_LEN(source.size())`, which it can be, right?

Comment (bzbarsky-apple) [reply examples/common/tracing/decoder/logging/ToCertificateString.cpp:43]: And also, we would write outside the `destination` buffer...

We can either not add 1 here but make sure we add 1 to the relevant size checks and allocations, or we can use separate variables for "length of the data" and "buffer needed to hold the data", or something else, but what we absolutely cannot do if overrun buffers.  That's how you get security bugs.

Comment (chshu) [reply examples/common/tracing/decoder/logging/ToCertificateString.cpp]: Fixed accordingly, thanks!

**Review by bzbarsky-apple — APPROVED** [review]
Yep, this makes much more sense, thank you!

**Review by andy31415 — APPROVED** [review]

**Review by dhrishi — APPROVED** [review]

**Comment by andy31415** [comment]
@chshu please fix restyle diffs

**Review by yufengwangca — APPROVED** [review]
